build.gradle

```diff
@@ -27,4 +27,13 @@ spotless {
     }
 }
 
+
+allprojects {
+    tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).configureEach {
+        kotlinOptions {
+            jvmTarget = JavaVersion.VERSION_1_8.toString()
```

Comment:
Is it a big deal that we are upgrading our JVM?

Reply:
Danny! Thank you for your patience - I was trying to investigate this to get you an intelligent answer. 

So to clarify I see that this comment was left near this code block
```groovy
allprojects {
    tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).configureEach {
        kotlinOptions {
            jvmTarget = JavaVersion.VERSION_1_8.toString()
```
but I know elsewhere I changed the CI to use the JDK 17, an upgrade from JDK 11.

**Short answer: yes, all of this should be fine**

**Longer answer below 👇**
For the sake of this discussion, there are three concepts that matter:
1. What version of Java Gradle is going to use to run the build pipeline
2. What version of the Java API and libraries we want to use in our source code
3. What version of Java byte code we want to compile our source code

Starting with #3, Android basically only supports Java 8 and lower, so we need to (and have been) set `targetCompatibility = Java 8`. 
Moving to #2, the version of the Java API that we want to use in our source code can be as low as we want but no higher than the version of Java bundled into the JDK that we are using (previously JDK 11, which has Java 11 but now upgraded to JDK 17, which has Java 17). The JDK will take care of compiling our Java 17 into Java 8 byte code.
Moving to #1, this is the version of the JDK that we have installed in our CI action or on our local machine. The reason that we need to use JDK 17 is actually because the Android Gradle Plugin version 8 specifying it needs JDK 17

So we end up with
* Android Gradle Plugin tells us to use JDK 17 so we install it in our CI and our local machines
* leads to Gradle using JDK 17 since that's what's available in CI / local machines
* Android devices run on Java 8 so we tell `targetCompatibility` to be Java 8
* Source code we limit to Java 8 mostly just because though it could go as high as Java 17 since we have JDK 17 installed

Lastly this block here is just to remind some tasks in Gradle that we are trying to target Java 8 byte code that's all
```groovy
allprojects {
    tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).configureEach {
        kotlinOptions {
            jvmTarget = JavaVersion.VERSION_1_8.toString()
```